addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.xml

```diff
@@ -4,85 +4,31 @@
     <t t-name="mail.messaging.component.ComposerTextInput" owl="1">
         <div class="o_ComposerTextInput">
             <t t-if="composer">
-                <textarea class="o_ComposerTextInput_textarea" t-esc="composer.textInputContent" placeholder="Write something..." t-on-keydown="_onKeydownTextarea" t-on-input="_onInputTextarea" t-ref="textarea" />
+                <div class="dropup o_ComposerTextInput_mentionDropdown" t-att-class="{ 'o-lowPosition': props.mentionsLowPosition }">
+                    <div class="o_ComposerTextInput_mentionDropdownPropositionList dropdown-menu" t-att-class="{ 'o-lowPosition': props.mentionsLowPosition, 'show': state.isMentionsOpen }">
+                        <t t-if="state.mentionSuggestions[0]" >
+                            <t t-foreach="state.mentionSuggestions[0]" t-as="mention" t-key="mention.id">
+                                <t t-if="mention.divider">
+                                    <div class="dropdown-divider" role="separator"/>
+                                </t>
+                                <a class="o_ComposerTextInput_mentionDropdownPropositionItem dropdown-item" t-att-class="{ 'active': mention.id === state.active_id }" href="#" role="menuitem" t-att-data-id="mention.id" t-on-click="_onClickMentionItem" t-on-mouseover="_onMouseOverMention">
+                                    <span t-att-class="mention.id ? 'o_ComposerTextInput_mentionDropdownPropositionItemImStatus' : ''" t-att-data-partner-id="mention.id">
+                                        <i t-if="mention.im_status == 'online'" class="o_ComposerTextInput_mentionDropdownPropositionItemIcon o-online fa fa-circle" title="Online" role="img" aria-label="User is online"/>
+                                        <i t-if="mention.im_status == 'away'" class="o_ComposerTextInput_mentionDropdownPropositionItemIcon o-idle fa fa-circle" title="Idle" role="img" aria-label="User is idle"/>
+                                        <i t-if="mention.im_status == 'offline'" class="o_ComposerTextInput_mentionDropdownPropositionItemIcon fa fa-circle-o" title="Offline" role="img" aria-label="User is offline"/>
+                                    </span>
+                                    <span class="o_ComposerTextInput_mentionDropdownPropositionItemName"><t t-esc="mention.name"/></span>
+                                    <t t-if="mention.email">
+                                        <span class="o_ComposerTextInput_mentionDropdownPropositionItemInfo">(<t t-esc="mention.email"/>)</span>
+                                    </t>
```

Comment:
this should be a sub-component, and I think it should be totally depended on entity `Partner` (and not another abstraction "mention" as here).
Basically mention suggestions (with `@` at least) are just (ordered) partners from relation with composer.

Reply:
In progress. Still need to style correctly the new component, especially ```PartnerImStatusIcon```

Reply:
Done